src/consumer/consumerGroup.js

```diff
@@ -220,13 +217,19 @@ module.exports = class ConsumerGroup {
       }
     }
 
-    this.memberAssignment = currentMemberAssignment
-    this.topics = keys(this.memberAssignment)
+    this.topics = currentMemberAssignment.map(({ topic }) => topic)
+    this.subscriptionState.assign(currentMemberAssignment)
     this.offsetManager = new OffsetManager({
       cluster: this.cluster,
       topicConfigurations: this.topicConfigurations,
       instrumentationEmitter: this.instrumentationEmitter,
-      memberAssignment: this.memberAssignment,
+      memberAssignment: currentMemberAssignment.reduce(
+        (partitionsByTopic, { topic, partitions }) => ({
+          ...partitionsByTopic,
+          [topic]: partitions,
+        }),
+        {}
+      ),
```

Comment:
Hmm, turns out that not only `SubscriptionState` needs to own member assignments, as the `OffsetManager` requires the same. It seems such a natural fit, as it really is subscription state, but perhaps it belongs to the `ConsumerGroup` after all? 🤔. Or perhaps instead, we should pass the result of `this.subscriptionState.assigned()` as a parameter to `OffsetManager` and have it accept the `Array<TopicPartitions>` type, dealing with it's own internal representation separately?

Reply:
Had another look at this, leaving me to think it's right the way it is, for now.

Naturally thinking, which partitions are assigned really is part of the domain of subscription state, where as for an offset manager it seems more like something to use. As far as I'm concerned, the ownership model is right this way.

Adapting `OffsetManager` to accept `Array<TopicPartitions>` is possible, but for the moving of code from `ConsumerGroup` to `OffsetManager`, we'd have to at least change a whole bunch of tests too, as the format of the `memberAssignment` parameter changes. If we'd actually want to eliminate this bit of code all together, we'd have to refactor all of OffsetManager in addition to the tests.